From 8837c5b485922bbe7e1cace5d3678f5b6f660286 Mon Sep 17 00:00:00 2001
From: Razvan Stefanescu <razvan.stefanescu@nxp.com>
Date: Mon, 25 Jun 2018 12:00:34 +0300
Subject: [PATCH 198/706] net: mscc: Force carrier state on port open/close

We use fixed link phys so avoid waiting for link availability before enabling
ports.

Signed-off-by: Razvan Stefanescu <razvan.stefanescu@nxp.com>
(cherry picked from commit 149783d419fd79f142057b9f4d9d65091701fa1d)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/mscc/ocelot.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/net/ethernet/mscc/ocelot.c b/drivers/net/ethernet/mscc/ocelot.c
index 808ddcf11d5b..b1ec24943566 100644
--- a/drivers/net/ethernet/mscc/ocelot.c
+++ b/drivers/net/ethernet/mscc/ocelot.c
@@ -379,6 +379,7 @@ void ocelot_port_adjust_link(struct net_device *dev)
 
 	if (!dev->phydev->link)
 		return;
+	netdev_err(dev, "DBG %s:%d\n", __func__, __LINE__);
 
 	/* Only full duplex supported for now */
 	ocelot_port_writel(port, DEV_MAC_MODE_CFG_FDX_ENA |
@@ -488,6 +489,8 @@ static int ocelot_port_open(struct net_device *dev)
 		netdev_err(dev, "Could not attach to PHY\n");
 		return err;
 	}
+	netif_carrier_on(dev);
+	ocelot->port_adjust_link(dev);
 
 	dev->phydev = port->phy;
 
@@ -500,6 +503,7 @@ static int ocelot_port_stop(struct net_device *dev)
 {
 	struct ocelot_port *port = netdev_priv(dev);
 
+	netif_carrier_off(dev);
 	phy_disconnect(port->phy);
 
 	dev->phydev = NULL;
-- 
2.17.1

